numpy/core/function_base.py

```diff
@@ -92,20 +92,36 @@ def linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None):
 
     if num <= 0:
         return array([], dtype)
+    if num == 1:
+        return array([start], dtype=dtype)
+    y = _nx.arange(0, num, dtype=dtype)
+    if endpoint:
+        num -= 1
+    y = y / float(num)
```

Comment:
this could be inplace too?

    y /= float(num)

won't make a big difference for division but it would be consistent with the following two lines